core/transaction-pool/graph/src/pool.rs

```diff
@@ -226,41 +174,51 @@ impl<B: ChainApi> Pool<B> {
 	/// To perform pruning we need the tags that each extrinsic provides and to avoid calling
 	/// into runtime too often we first lookup all extrinsics that are in the pool and get
 	/// their provided tags from there. Otherwise we query the runtime at the `parent` block.
-	pub fn prune(&self, at: &BlockId<B::Block>, parent: &BlockId<B::Block>, extrinsics: &[ExtrinsicFor<B>]) -> Result<(), B::Error> {
-		let mut tags = Vec::with_capacity(extrinsics.len());
+	pub fn prune(
+		&self,
+		at: &BlockId<B::Block>,
+		parent: &BlockId<B::Block>,
+		extrinsics: &[ExtrinsicFor<B>],
+	) -> impl Future<Output=Result<(), B::Error>> {
```

Comment:
Couldn't we relax the lifetime in the returned future and avoid cloning `self.data` (and keeping it in an `Arc`)?

Reply:
It needed to be `Box<>` .with `'static` lifetime to be spawned in executor' threads.